Mathlib/Order/Filter/Basic.lean

```diff
@@ -306,6 +306,38 @@ theorem iInf_eq_generate (s : ι → Filter α) : iInf s = generate (⋃ i, (s i
 theorem mem_iInf_of_mem {f : ι → Filter α} (i : ι) {s} (hs : s ∈ f i) : s ∈ ⨅ i, f i :=
   iInf_le f i hs
 
+@[elab_as_elim]
+theorem iInf_sets_induct {f : ι → Filter α} {s : Set α} (hs : s ∈ iInf f) {p : Set α → Prop}
```

Comment:
It took a while but I think I get the proof: we want to show `p` holds, but its conditions are a bit weird and asymmetric in `ins`. So instead we build a slightly different predicate `q` on sets which implies `p` but also forms a filter `𝓖`. Then any set in the infimum of `f` can be shown to be an element of `𝓖` and therefore also satisfy `p`. Did I get that right? Could you add some comments explaining this?